### PR TITLE
fix: git branch protection fixes

### LINKED ISSUE #28056: [Task] Frontend changes for branch protection
- Add settings for branch protection
- Protected screen for locked branches

### PR DESCRIPTION
**Pull request by brayn003**
## Description
- Adds license flag for branch protection
- Disables omni-bar and keyboard shortcuts for protected branches
- Adds navigation for multiple pages
- Hides preview on protected views
- Adds feature flag for Git Connection Success

#### PR fixes following issue(s)
Fixes #28056 

#### Media
image

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

- [x] Manual
- [ ] JUnit
- [ ] Jest
- [x] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx [hunk 1]
```diff
@@ -91,6 +98,72 @@ function ConnectionSuccess() {
...
+    return (
+      <Text renderAs="p">{createMessage(GIT_CONNECT_SUCCESS_MESSAGE)}</Text>
+    );
+  };
+
+  const postBranchProtectionContent = () => {
+    return (
+      <>
+        <Text renderAs="p" style={{ marginBottom: 16 }}>
+          Right now,{" "}
```
Comment (dipyamanbiswas07) [inline app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx:111]: {" "} is this extra space required?

File: app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx [hunk 2]
```diff
@@ -91,6 +98,72 @@ function ConnectionSuccess() {
...
+          <Tag isClosable={false} style={{ display: "inline-flex" }}>
```
Comment (dipyamanbiswas07) [inline app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx:112]: can we not use in line styles please

File: app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx [hunk 3]
```diff
@@ -91,6 +98,72 @@ function ConnectionSuccess() {
...
+            {defaultBranchName}
+          </Tag>{" "}
+          is set as the default branch and it is protected.
+        </Text>
+        <Text renderAs="p" style={{ marginBottom: 8 }}>
```
Comment (dipyamanbiswas07) [inline app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx:117]: can we not use in line styles please

File: app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx [hunk 4]
```diff
@@ -91,6 +98,72 @@ function ConnectionSuccess() {
...
+        </FeatureList>
+        <Text>{createMessage(BRANCH_PROTECTION_CHANGE_RULE)}</Text>
+      </>
+    );
+  };
+
+  const preBranchProtectionActions = () => {
+    return (
+      <Button
+        data-testid="t--start-using-git-button"
```
Comment (dipyamanbiswas07) [inline app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx:140]: both this button and the one below use the same data-testid, should we separate them?

File: app/client/src/pages/Editor/gitSync/Tabs/GitSettings/GitProtectedBranches.tsx [hunk 5]
```diff
@@ -86,17 +88,19 @@ function GitProtectedBranches() {
...
-          >
-            {createMessage(APPSMITH_ENTERPRISE)}
-          </Link>
-        </SectionDesc>
+        {!isGitProtectedFeatureLicensed && (
+          <SectionDesc kind="body-m" renderAs="p">
+            To protect multiple branches, try{" "}
+            <Link
+              kind="primary"
+              style={{ display: "inline-flex" }}
```
Comment (dipyamanbiswas07) [inline app/client/src/pages/Editor/gitSync/Tabs/GitSettings/GitProtectedBranches.tsx:96]: can we not use in line styles please

File: app/client/src/pages/Editor/gitSync/Tabs/GitSettings/GitProtectedBranches.tsx [hunk 6]
```diff
@@ -86,17 +88,19 @@ function GitProtectedBranches() {
...
+              target="_blank"
+              to="https://www.appsmith.com/enterprise?lead_source=git%20feat%20branch%20config"
```
Comment (dipyamanbiswas07) [inline app/client/src/pages/Editor/gitSync/Tabs/GitSettings/GitProtectedBranches.tsx:98]: this should come from a constant, also should we redirect to pricing page/customer portal instead?

File: app/client/src/selectors/gitSyncSelectors.tsx [hunk 7]
```diff
@@ -244,3 +244,8 @@ export const getIsGitProtectedFeatureEnabled = createSelector(
   selectFeatureFlags,
   (flags) => !!flags?.release_git_branch_protection_enabled,
 );
+
+export const getIsGitProtectedFeatureLicensed = createSelector(
```
Comment (dipyamanbiswas07) [inline app/client/src/selectors/gitSyncSelectors.tsx:248]: can we please use planHelpers.tsx for license_ feature flag checks?

File: app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx (hunk 1 above)
Comment (brayn003) [reply app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx:111]: yes

File: app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx (hunk 4 above)
Comment (brayn003) [reply app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx:140]: Both of them are conditionally rendered. At a time only one of those buttons is used. Also, need this so it doesn't break cypress tests.

Comment (dipyamanbiswas07) [reply app/client/src/pages/Editor/gitSync/Tabs/ConnectionSuccess.tsx:150]: Alright

**Comment by Parthvi12** [comment]
@brayn003 Can you fix following bugs on the DP:

- [ ] On creating new branch, the commit input box is disabled for the first time

https://github.com/appsmithorg/appsmith/assets/80334441/4e97d292-c62b-430d-8e0d-c6a7dd18a7a9

- [ ] Able to import template in protected branch, in protected mode

https://github.com/appsmithorg/appsmith/assets/80334441/37868be8-5766-4c3d-bab1-d68b8a728a8b

**Review by dipyamanbiswas07 — DISMISSED** [review]
LGTM

**Pull request merged**